Cannot build net7.0 with FSharp.Compiler.Service 42.7.100

# Cannot build net7.0 with FSharp.Compiler.Service 42.7.100

**Repro steps**

With the latest Visual Studio 17.4 and its .NET 7 SDK and templates,
in a new folder with the test project name invoke:

```
dotnet new console -lang f#
```

It creates the net7.0 console app F# project.
Then add the latest FSharp.Compiler.Service 42.7.100:

```
dotnet add package FSharp.Compiler.Service --version 42.7.100
```

It fails:

```
> dotnet add package FSharp.Compiler.Service --version 42.7.100
  Determining projects to restore...
  Writing C:\TEMP\TEMP\tmp1BB9.tmp
info : X.509 certificate chain validation will use the default trust store selected by .NET.
info : X.509 certificate chain validation will use the default trust store selected by .NET.
info : Adding PackageReference for package 'FSharp.Compiler.Service' into project 'C:\tmp\_221109_0646\_221109_0646.fsproj'.
info : Restoring packages for C:\tmp\_221109_0646\_221109_0646.fsproj...
info :   CACHE https://api.nuget.org/v3-flatcontainer/microsoft.build.framework/index.json
info :   CACHE https://api.nuget.org/v3-flatcontainer/microsoft.build.tasks.core/index.json
info :   CACHE https://api.nuget.org/v3-flatcontainer/microsoft.build.utilities.core/index.json
error: NU1102: Unable to find package Microsoft.Build.Framework with version (>= 17.4.0-preview-22469-04)
error:   - Found 42 version(s) in nuget.org [ Nearest version: 17.3.2 ]
error:   - Found 0 version(s) in MyNuGet
error:   - Found 0 version(s) in Microsoft Visual Studio Offline Packages
error:   - Found 0 version(s) in C:\Program Files\dotnet\sdk\7.0.100\FSharp\library-packs
error: NU1102: Unable to find package Microsoft.Build.Tasks.Core with version (>= 17.4.0-preview-22469-04)
error:   - Found 42 version(s) in nuget.org [ Nearest version: 17.3.2 ]
error:   - Found 0 version(s) in MyNuGet
error:   - Found 0 version(s) in Microsoft Visual Studio Offline Packages
error:   - Found 0 version(s) in C:\Program Files\dotnet\sdk\7.0.100\FSharp\library-packs
error: NU1102: Unable to find package Microsoft.Build.Utilities.Core with version (>= 17.4.0-preview-22469-04)
error:   - Found 42 version(s) in nuget.org [ Nearest version: 17.3.2 ]
error:   - Found 0 version(s) in MyNuGet
error:   - Found 0 version(s) in Microsoft Visual Studio Offline Packages
error:   - Found 0 version(s) in C:\Program Files\dotnet\sdk\7.0.100\FSharp\library-packs
error: Package 'FSharp.Compiler.Service' is incompatible with 'all' frameworks in project 'C:\tmp\_221109_0646\_221109_0646.fsproj'.
```

To work around, add this reference manually to the project file:

```xml
  <ItemGroup>
    <PackageReference Include="FSharp.Compiler.Service" Version="42.7.100" />
  </ItemGroup>
```

and build

```
dotnet build
```

It fails due to not found `Microsoft.Build.*` packages:

```
> dotnet build
MSBuild version 17.4.0+18d5aef85 for .NET
  Determining projects to restore...
C:\tmp\_221109_0646\_221109_0646.fsproj : error NU1102: Unable to find package Microsoft.Build.Framework with version (>= 17.4.0-preview-22469-04)
C:\tmp\_221109_0646\_221109_0646.fsproj : error NU1102:   - Found 42 version(s) in nuget.org [ Nearest version: 17.3.2 ]
C:\tmp\_221109_0646\_221109_0646.fsproj : error NU1102:   - Found 0 version(s) in MyNuGet
C:\tmp\_221109_0646\_221109_0646.fsproj : error NU1102:   - Found 0 version(s) in Microsoft Visual Studio Offline Packages
C:\tmp\_221109_0646\_221109_0646.fsproj : error NU1102:   - Found 0 version(s) in C:\Program Files\dotnet\sdk\7.0.100\FSharp\library-packs
C:\tmp\_221109_0646\_221109_0646.fsproj : error NU1102: Unable to find package Microsoft.Build.Tasks.Core with version (>= 17.4.0-preview-22469-04)
C:\tmp\_221109_0646\_221109_0646.fsproj : error NU1102:   - Found 42 version(s) in nuget.org [ Nearest version: 17.3.2 ]
C:\tmp\_221109_0646\_221109_0646.fsproj : error NU1102:   - Found 0 version(s) in MyNuGet
C:\tmp\_221109_0646\_221109_0646.fsproj : error NU1102:   - Found 0 version(s) in Microsoft Visual Studio Offline Packages
C:\tmp\_221109_0646\_221109_0646.fsproj : error NU1102:   - Found 0 version(s) in C:\Program Files\dotnet\sdk\7.0.100\FSharp\library-packs
C:\tmp\_221109_0646\_221109_0646.fsproj : error NU1102: Unable to find package Microsoft.Build.Utilities.Core with version (>= 17.4.0-preview-22469-04)
C:\tmp\_221109_0646\_221109_0646.fsproj : error NU1102:   - Found 42 version(s) in nuget.org [ Nearest version: 17.3.2 ]
C:\tmp\_221109_0646\_221109_0646.fsproj : error NU1102:   - Found 0 version(s) in MyNuGet
C:\tmp\_221109_0646\_221109_0646.fsproj : error NU1102:   - Found 0 version(s) in Microsoft Visual Studio Offline Packages
C:\tmp\_221109_0646\_221109_0646.fsproj : error NU1102:   - Found 0 version(s) in C:\Program Files\dotnet\sdk\7.0.100\FSharp\library-packs
  Failed to restore C:\tmp\_221109_0646\_221109_0646.fsproj (in 271 ms).

Build FAILED.
```

**Expected behavior**

All the above commands should work and build an app, as a result.

**Actual behavior**

Some commands fail as described.

**Known workarounds**

None.

**Related information**

* Operating system - Windows 10
* .NET Runtime kind - net7.0
* Editing Tools - Visual Studio 17.4

--- a/eng/Versions.props
+++ b/eng/Versions.props
@@ -100,7 +100,7 @@
     <MicrosoftVisualStudioShellPackagesVersion>17.4.0-preview-3-32916-145</MicrosoftVisualStudioShellPackagesVersion>
     <VisualStudioProjectSystemPackagesVersion>17.4.342-pre</VisualStudioProjectSystemPackagesVersion>
     <MicrosoftVisualStudioThreadingPackagesVersion>17.4.23-alpha</MicrosoftVisualStudioThreadingPackagesVersion>
-    <MicrosoftBuildOverallPackagesVersion>17.3.2</MicrosoftBuildOverallPackagesVersion>
+    <MicrosoftBuildOverallPackagesVersion>17.4.0</MicrosoftBuildOverallPackagesVersion>
     <!-- Roslyn packages -->
     <MicrosoftCodeAnalysisEditorFeaturesVersion>$(RoslynVersion)</MicrosoftCodeAnalysisEditorFeaturesVersion>
     <MicrosoftCodeAnalysisEditorFeaturesTextVersion>$(RoslynVersion)</MicrosoftCodeAnalysisEditorFeaturesTextVersion>

--- a/eng/Versions.props
+++ b/eng/Versions.props
@@ -33,7 +33,7 @@
     <!-- FSharp.Compiler.Service version -->
     <FCSMajorVersion>42</FCSMajorVersion>
     <FCSMinorVersion>7</FCSMinorVersion>
-    <FCSBuildVersion>100</FCSBuildVersion>
+    <FCSBuildVersion>101</FCSBuildVersion>
     <FCSRevisionVersion>$(FSRevisionVersion)</FCSRevisionVersion>
     <FSharpCompilerServicePackageVersion>$(FCSMajorVersion).$(FCSMinorVersion).$(FCSBuildVersion)</FSharpCompilerServicePackageVersion>
     <FSharpCompilerServiceVersion>$(FCSMajorVersion).$(FCSMinorVersion).$(FCSBuildVersion).$(FCSRevisionVersion)</FSharpCompilerServiceVersion>
@@ -100,7 +100,7 @@
     <MicrosoftVisualStudioShellPackagesVersion>17.4.0-preview-3-32916-145</MicrosoftVisualStudioShellPackagesVersion>
     <VisualStudioProjectSystemPackagesVersion>17.4.342-pre</VisualStudioProjectSystemPackagesVersion>
     <MicrosoftVisualStudioThreadingPackagesVersion>17.4.23-alpha</MicrosoftVisualStudioThreadingPackagesVersion>
-    <MicrosoftBuildOverallPackagesVersion>17.4.0-preview-22469-04</MicrosoftBuildOverallPackagesVersion>
+    <MicrosoftBuildOverallPackagesVersion>17.3.2</MicrosoftBuildOverallPackagesVersion>
     <!-- Roslyn packages -->
     <MicrosoftCodeAnalysisEditorFeaturesVersion>$(RoslynVersion)</MicrosoftCodeAnalysisEditorFeaturesVersion>
     <MicrosoftCodeAnalysisEditorFeaturesTextVersion>$(RoslynVersion)</MicrosoftCodeAnalysisEditorFeaturesTextVersion>

